File: packages/ogdf.js/src/layouts/planarity/PlanarizationGridLayout.js
import { PARAMETER_TYPE } from '../../utils/parameters'
import CrossingMinimizationModule from '../../module/config/CrossingMinimization'
import CCLayoutPackModule from '../../module/config/CCLayoutPack'
import GridLayoutPlanRepModule from '../../module/config/GridLayoutPlanRep'

export default {
    pageRatio: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [0, Infinity],
        default: 1.0
    },
    separation: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [0, Infinity],
        default: 20.0
    },
    crossMin: {
        type: PARAMETER_TYPE.MODULE,
        module: CrossingMinimizationModule,
        default: CrossingMinimizationModule.SubgraphPlanarizer
    },
    packer: {
        type: PARAMETER_TYPE.MODULE,
        module: CCLayoutPackModule,
        default: CCLayoutPackModule.TileToRowsCCPacker
    },
    planarLayouter: {
        type: PARAMETER_TYPE.MODULE,
        module: GridLayoutPlanRepModule,
        default: GridLayoutPlanRepModule.MixedModelLayout
    }
}

File: packages/ogdf.js/src/entry/planarity/PlanarizationGridLayout.cpp
#include <ogdf/planarity/PlanarizationGridLayout.h>
#include "../main.h"

EM_PORT_API(PlanarizationGridLayout *)
LayoutModule_PlanarizationGridLayout(double pageRatio, double separation, CrossingMinimizationModule *crossMin, CCLayoutPackModule *packer, GridLayoutPlanRepModule *planarLayouter)
{
	//LayoutModule
	PlanarizationGridLayout *model = new PlanarizationGridLayout();

	model->pageRatio(pageRatio);
	model->separation(separation);

	model->setCrossMin(crossMin);
	model->setPacker(packer);
	model->setPlanarLayouter(planarLayouter);

	return model;
}